### ISSUE TITLE
YOLO v4 Error: Floating point exception (core dumped)

### ISSUE THREAD

**Issue by awaisbajwaml** [AlexeyAB/darknet#5678]
@AlexeyAB and team

Thanks for the wonderful work and keeping the lights on and brighter with the new v4 . Need some help or direction here:

I am trying to train my custom object on Tesla T4 with the following specs:

![oom](https://user-images.githubusercontent.com/47017344/82411398-b5af4080-9a3f-11ea-871d-4361e09bd27b.png)

When I train my dataset I get the following error.

![FloatingPoint](https://user-images.githubusercontent.com/47017344/82411406-b8aa3100-9a3f-11ea-8dc0-6c3187fea69c.png)

Just to give you a bit of history, I got OOM errors and I had to reduce the batchsize, subdivisions and height/width to fix the OOM errors.

Here is the config file :  

[yolo-obj.txt](https://github.com/AlexeyAB/darknet/files/4654949/yolo-obj.txt)

Any help is appreciated.

**Comment by AlexeyAB** [comment]
* What command do you use?
* Do you use the latest Darknet version?
* Show content of files bad.list and bad_label.list
* Show such screenshot:
![image](https://user-images.githubusercontent.com/4096485/82433618-61857a00-9a9a-11ea-9bde-2c7f4d94ba1a.png)

**Comment by awaisbajwaml** [comment]
@AlexeyAB  here are the details 



I use this command

` ./darknet detector train yolo-files/obj.data yolo-files/yolo-obj.cfg yolov4.conv.137 -dont_show`

yes, I use the latest Darknet version. cloned this:

![yolov4](https://user-images.githubusercontent.com/47017344/82450100-41de5980-9a7a-11ea-842c-125e1f386ad6.png)


I could not find any file name bad.list or bad_llabel.list anywhere on my VM.



Here is the screen shot:

![output_screen](https://user-images.githubusercontent.com/47017344/82450113-460a7700-9a7a-11ea-96de-6c2b245b16f5.png)


I appreciate your help here.

**Comment by AlexeyAB** [comment]
Can you run it successfully by using previous version of darknet, f.e. Commits on May 11, 2020 ?

Or by using YOLOv4 pre-release: https://github.com/AlexeyAB/darknet/archive/darknet_yolo_v4_pre.zip

**Comment by awaisbajwaml** [comment]
@AlexeyAB 

yes sure,  I will build again using pre-release that you provided and can post the comments.

**Comment by awaisbajwaml** [comment]
@AlexeyAB 

I tried the following 

1- Used the pre-release here https://github.com/AlexeyAB/darknet/archive/darknet_yolo_v4_pre.zip

Same Error.

2- Build the darkent again with Opencv=0 , just an option to see if opencv is a problem. 



Same Error.

3- Changed batch and subdivision to many different options 

Same Error.

4- Changed the learning rate and decrease it to 0.001 

Same Error.

Here is the last state with the config file.  

network init

![init](https://user-images.githubusercontent.com/47017344/82525267-f15d0f80-9afe-11ea-8577-22c947e5029a.png)


![end](https://user-images.githubusercontent.com/47017344/82525288-fae67780-9afe-11ea-817d-17b8c97da56a.png)


Here is the modified config file.

[yolo-obj.txt](https://github.com/AlexeyAB/darknet/files/4660672/yolo-obj.txt)


Is there any way we can enable the training log and it can tell us the line number where the multiplication or division is failing. ? 


One other option I was thinking is to use YOLO3 to see if it works but not sure if I go that route. 


Please suggest.

**Comment by AlexeyAB** [comment]
* What OS do you use?

* Put this file https://github.com/AlexeyAB/darknet/blob/master/3rdparty/pthreads/bin/pthreadVC2.dll?raw=true near with `darknet.exe`

* Try to keep only 1 image in your training dataset and run training.

**Comment by awaisbajwaml** [comment]
@AlexeyAB 

I am using Linux OS:

Ubuntu 18.04

**Comment by awaisbajwaml** [comment]
@AlexeyAB 

do you have the same pthread file for Linux?

**Comment by awaisbajwaml** [comment]
@AlexeyAB 

I am able to solve the issue. Although I re-installed CUDA and cuDNN twice and did some verification on CUDA compatibility and GPU itself. 

But in the end, it was a data issue, after fixing the training data files I was able to start the training.

Closing this issue.

**Comment by henryfhq-drone** [comment]
@awaisbajwaml Hi, I encountered the same issue and I'm using the same system as you. I try looking through the settings and the training data and it doesn't help. Can you elaborate on what do you mean by fixing the data training files? 

Thanks a lot
